File: Lib/RelationSchema.cs
using System;
using System.Linq;

namespace System.Data.EntitySchema {

  public class RelationSchema {

    public string Name = "";

    public string PrimaryEntityName = "";
    public string ForeignEntityName = "";
    public string ForeignKeyIndexName = "";

    public bool IsLookupRelation = true;
    public bool PrimaryEntityIsOptional = false;
    public bool ForeignEntityIsMultiple = true;

    public string PrimaryNavigationName = "";
    public string ForeignNavigationName = "";
    public string PrimaryNavigationSummary = "";
    public string ForeignNavigationSummary = "";

    public bool CascadeDelete = false;

  }

}

File: Lib_JS/src/relationSchema.ts

export class RelationSchema {

  public name: string = '';

  public primaryEntityName: string = '';
  public foreignEntityName: string = '';
  public foreignKeyIndexName: string = '';

  public isLookupRelation: boolean = true;
  public primaryEntityIsOptional: boolean = false;
  public foreignEntityIsMultiple: boolean = true;

  public primaryNavigationName: string = '';
  public foreignNavigationName: string = '';
  public primaryNavigationSummary: string = '';
  public foreignNavigationSummary: string = '';

  public cascadeDelete: boolean = false;

}
